File: google-paths/foundational/codingbat/python/interpret.py
# http://codingbat.com/prob/p234011
def interpret(value, commands, args):
  ret = 0
  for index, command in enumerate(commands):
    if (command == '+'):
      if (index == 0):
        ret = value + args[index]
      else:
        ret += args[index]
    elif (command == '-'):
      if (index == 0):
        ret = value - args[index]
      else:
        ret -= args[index]
    elif (command == '*'):
      if (index == 0):
        ret = value * args[index]
      else:
        ret *= args[index]
    elif (command == '/'):
      if (index == 0):
        ret = value / args[index]
      else:
        ret /= args[index]
    else:
      return -1
  return ret

File: google-paths/foundational/codingbat/java/interpret.java
// http://codingbat.com/prob/p294185
public int interpret(int value, String[] commands, int[] args) {
  int result = 0;
  int argIndex = 0;
  for (String c : commands) {
    switch (c) {
      case "+":
        if (argIndex == 0)
          result = value + args[argIndex];
        else
          result += args[argIndex];
        break;
      case "-":
        if (argIndex == 0)
          result = value - args[argIndex];
        else
          result -= args[argIndex];
        break;
      case "*":
        if (argIndex == 0)
          result = value * args[argIndex];
        else
          result *= args[argIndex];
        break;
      default:
        return -1;
    }
    argIndex++;
  }

  return result;
}
